xaitk_saliency/utils/bin/sal_on_coco_dets.py

```diff
@@ -94,37 +99,52 @@ def det_sal_on_img_dir(
     if verbose:
         logging.basicConfig(level=logging.INFO)
 
-    sal_maps_list, ref_dets_list = gen_detector_sal(
-        imgs,
-        blackbox_detector,
-        img_perturber,
-        sal_generator,
-        verbose=verbose
-    )
+    try:
+        sal_maps = gen_coco_sal(
+            dets_dset,
+            blackbox_detector,
+            img_perturber,
+            sal_generator,
+            verbose=verbose
+        )
+    except MismatchedLabelsError as e:
+        print(str(e))
+        exit()
```

Comment:
Why not just let python's exception handling print the stack-trace and error message for us? This exit will return a 0-code which can mislead callers to thinking the tool exited "successfully."